source/Posets/Delta-V-CompletePoset.lagda

```diff
@@ -0,0 +1,854 @@
+open import UF.Equiv
+open import UF.Retracts
+open import UF.Subsingletons-FunExt
+open import UF.NotNotStablePropositions
+open import UF.Embeddings
+open import UF.Sets
+open import UF.ExcludedMiddle
+open import Slice.Family
+
+module Posets.Delta-V-CompletePoset
+ (pt : propositional-truncations-exist)
+ (fe : Fun-Ext)
+ (pe : Prop-Ext)
+  where
+
+open import Locales.Frame pt fe hiding (𝟚)
+open import Posets.TwoElementPoset pt fe
+open AllCombinators pt fe
+
+module δ-complete-poset {𝓤 𝓦 : Universe} (𝓥 : Universe) (A : Poset 𝓤 𝓦) where
+
+ δ : (x y : ∣ A ∣ₚ) → (P : Ω 𝓥) → (𝟙{𝓥} + P holds) → ∣ A ∣ₚ 
+ δ x y P (inl _) = x
+ δ x y P (inr _) = y
+
+ _≤_ : ∣ A ∣ₚ → ∣ A ∣ₚ → Ω 𝓦
+ _≤_ = rel-syntax A
+
+ open Joins (_≤_)
+
+ δ-fam : (x y : ∣ A ∣ₚ) → (P : Ω 𝓥) → Fam 𝓥 ∣ A ∣ₚ
+ δ-fam x y P = ((𝟙 + P holds) , δ x y P)
+
+ is-δ-complete : 𝓤 ⊔ 𝓦 ⊔ (𝓥 ⁺) ̇ 
+ is-δ-complete = (x y : ∣ A ∣ₚ)
+               → (x ≤ y) holds
+               → (P : Ω 𝓥)
+               → Σ s ꞉ ∣ A ∣ₚ , (s is-lub-of (δ-fam x y P)) holds
+
+ sup-of-δ : is-δ-complete
+          → (x y : ∣ A ∣ₚ)
+          → (x ≤ y) holds
+          → (P : Ω 𝓥)
+          → ∣ A ∣ₚ
+ sup-of-δ i x y o P = pr₁ (i x y o P)
+
+ module _ (i : is-δ-complete)
+          (x y : ∣ A ∣ₚ)
+          (o : (x ≤ y) holds)
+          (P : Ω 𝓥)
+           where
+
+  is-sup-of-δ : ((sup-of-δ i x y o P) is-lub-of (δ-fam x y P)) holds
+  is-sup-of-δ = pr₂ (i x y o P)
+
+  is-ub-of-δ : ((sup-of-δ i x y o P) is-an-upper-bound-of (δ-fam x y P)) holds
+  is-ub-of-δ = pr₁ is-sup-of-δ
+
+  has-lub-cond-δ : ((u , _) : upper-bound (δ-fam x y P))
+                 → ((sup-of-δ i x y o P) ≤ u) holds
+  has-lub-cond-δ = pr₂ is-sup-of-δ
+
+  lower-＝-sup-δ : ¬ (P holds) → x ＝ sup-of-δ i x y o P
+  lower-＝-sup-δ not-P = ≤-is-antisymmetric A x-≤-sup sup-≤-x
+   where
+    x-≤-sup : (x ≤ sup-of-δ i x y o P) holds
+    x-≤-sup = is-ub-of-δ (inl ⋆)
+
+    x-is-ub : (x is-an-upper-bound-of (δ-fam x y P)) holds
+    x-is-ub (inl ⋆) = ≤-is-reflexive A x
+    x-is-ub (inr in-P) = 𝟘-induction (not-P in-P)
+
+    sup-≤-x : (sup-of-δ i x y o P ≤ x) holds
+    sup-≤-x = has-lub-cond-δ (x , x-is-ub)
+
+  upper-＝-sup-δ : P holds → y ＝ sup-of-δ i x y o P
+  upper-＝-sup-δ in-P = ≤-is-antisymmetric A y-≤-sup sup-≤-y
+   where
+    y-≤-sup : (y ≤ sup-of-δ i x y o P) holds
+    y-≤-sup = is-ub-of-δ (inr in-P)
+
+    y-is-ub : (y is-an-upper-bound-of (δ-fam x y P)) holds
+    y-is-ub (inl ⋆) = o
+    y-is-ub (inr in-P) = ≤-is-reflexive A y
+
+    sup-≤-y : (sup-of-δ i x y o P ≤ y) holds
+    sup-≤-y = has-lub-cond-δ (y , y-is-ub)
+   
+  sup-δ-≤-upper : (sup-of-δ i x y o P ≤ y) holds
+  sup-δ-≤-upper = has-lub-cond-δ (y , y-is-ub)
+   where
+    y-is-ub : (y is-an-upper-bound-of (δ-fam x y P)) holds
+    y-is-ub (inl ⋆) = o
+    y-is-ub (inr _) = ≤-is-reflexive A y
+
+\end{code}
+
+Future work: I would be nice to show that classically every poset is δ_𝓥
+complete and to provide the critical examples of δ_𝓥 complete posets (𝓥-sup
```

Comment:
Could you label this with a `TODO` please? This makes it easier to search (`grep`) all code for future work.